### DISCUSSION TITLE
errors in configure.ac

### DISCUSSION THREAD

**Discussion by the-loaf** [hengband/hengband#3545] [General]
Hello! I'm trying to install hengband-3.0.0.87-Alpha. I tried autoconf and it throws two errors:

$ autoconf
configure.ac:7: error: possibly undefined macro: AM_INIT_AUTOMAKE
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
configure.ac:17: error: possibly undefined macro: AM_CONDITIONAL
configure.ac:52: error: possibly undefined macro: AC_DEFINE

then when I try to use configure:

./configure --disable-japanese
./configure: line 2739: AM_INIT_AUTOMAKE: command not found
./configure: line 2753: syntax error near unexpected token `SET_GID,'
./configure: line 2753: `AM_CONDITIONAL(SET_GID, test "$GAMEGROUP" != "")'

I'm running Xubuntu 22.04.2

**Comment by tanguband** [comment]
Running ./bootstrap before ./configure might work

**Reply by the-loaf** [reply]
I tried that:

$ ./bootstrap 
+ aclocal -I m4
+ autoheader
+ automake --foreign --add-missing --copy
configure.ac:100: installing './compile'
configure.ac:7: installing './install-sh'
configure.ac:7: installing './missing'
configure.ac:164: error: required file 'lib/xtra/Makefile.in' not found
configure.ac:164: error: required file 'lib/xtra/graf/Makefile.in' not found
configure.ac:164: error: required file 'lib/xtra/music/Makefile.in' not found
configure.ac:164: error: required file 'lib/xtra/sound/Makefile.in' not found
src/Makefile.am: installing './depcomp'
+ autoconf
configure.ac:52: error: possibly undefined macro: AC_DEFINE
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.

$ ./configure --disable-japanese
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking for g++... g++
checking whether the C++ compiler works... yes
checking for C++ compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C++... yes
checking whether g++ accepts -g... yes
checking for g++ option to enable C++11 features... none needed
checking whether make supports the include directive... yes (GNU style)
checking dependency style of g++... gcc3
checking whether g++ supports C++20 features with -std=gnu++20... yes
./configure: line 5308: PKG_PROG_PKG_CONFIG: command not found
checking for initscr in -lncursesw... yes
checking for iconv_open in -liconv... no
./configure: line 5598: syntax error near unexpected token `libcurl,'
./configure: line 5598: `  PKG_CHECK_MODULES(libcurl, libcurl)'

**Reply by habu1010** [reply]
Install pkg-config and libcurl4-openssl-dev with apt-get.

**Reply by the-loaf** [reply]
Getting closer:
$ ./bootstrap 
+ aclocal -I m4
+ autoheader
+ automake --foreign --add-missing --copy
configure.ac:164: error: required file 'lib/xtra/Makefile.in' not found
configure.ac:164: error: required file 'lib/xtra/graf/Makefile.in' not found
configure.ac:164: error: required file 'lib/xtra/music/Makefile.in' not found
configure.ac:164: error: required file 'lib/xtra/sound/Makefile.in' not found
+ autoconf

$ ./configure --disable-japanese
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking for g++... g++
checking whether the C++ compiler works... yes
checking for C++ compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C++... yes
checking whether g++ accepts -g... yes
checking for g++ option to enable C++11 features... none needed
checking whether make supports the include directive... yes (GNU style)
checking dependency style of g++... gcc3
checking whether g++ supports C++20 features with -std=gnu++20... yes
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for initscr in -lncursesw... yes
checking for iconv_open in -liconv... no
checking for libcurl... yes
checking for gcc... gcc
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... gcc3
checking how to run the C++ preprocessor... g++ -std=gnu++20 -E
checking for X... no
checking for md5sum... /usr/bin/md5sum
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for fcntl.h... yes
checking for sys/file.h... yes
checking for sys/ioctl.h... yes
checking for sys/time.h... yes
checking for termio.h... yes
checking for unistd.h... (cached) yes
checking for stdint.h... (cached) yes
checking for an ANSI C-conforming const... yes
checking for pid_t... yes
checking for size_t... yes
checking whether struct tm is in sys/time.h or time.h... time.h
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking whether gcc needs -traditional... no
checking for working memcmp... yes
checking for strftime... yes
checking for vprintf... yes
checking for gethostname... yes
checking for mkdir... yes
checking for select... yes
checking for socket... yes
checking for strtol... yes
checking for mkstemp... yes
checking for usleep... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: error: cannot find input file: `Makefile.in'

$ ls -l lib/xtra/
total 0

**Reply by tanguband** [reply]
lib/xtra requires git submodule, so the Mac version build description page may be helpful
https://backwardseric.github.io/hengband/HengbandSourceCode.html

**Reply by tanguband** [reply]
You probably also need libiconv
